### ISSUE TITLE
Rails 6.1, undefined method `event_subscription' for Bugsnag::Railtie:Class

### ISSUE THREAD

**Issue by xxx44yyy** [bugsnag/bugsnag-ruby#653]
```
Uncaught exception: undefined method `event_subscription' for Bugsnag::Railtie:Class
	/private/var/www/apps/my_project/shared/bundle/ruby/2.7.0/gems/railties-6.1.0/lib/rails/railtie.rb:209:in `method_missing'
	/private/var/www/apps/my_project/shared/bundle/ruby/2.7.0/gems/bugsnag-6.19.0/lib/bugsnag/integrations/railtie.rb:44:in `block (2 levels) in <class:Railtie>'
	/private/var/www/apps/my_project/shared/bundle/ruby/2.7.0/gems/bugsnag-6.19.0/lib/bugsnag/integrations/railtie.rb:44:in `each'
	/private/var/www/apps/my_project/shared/bundle/ruby/2.7.0/gems/bugsnag-6.19.0/lib/bugsnag/integrations/railtie.rb:44:in `block in <class:Railtie>'
	/private/var/www/apps/my_project/shared/bundle/ruby/2.7.0/gems/activesupport-6.1.0/lib/active_support/lazy_load_hooks.rb:68:in `block in execute_hook'
	/private/var/www/apps/my_project/shared/bundle/ruby/2.7.0/gems/activesupport-6.1.0/lib/active_support/lazy_load_hooks.rb:61:in `with_execution_control'
	/private/var/www/apps/my_project/shared/bundle/ruby/2.7.0/gems/activesupport-6.1.0/lib/active_support/lazy_load_hooks.rb:66:in `execute_hook'
	/private/var/www/apps/my_project/shared/bundle/ruby/2.7.0/gems/activesupport-6.1.0/lib/active_support/lazy_load_hooks.rb:43:in `block in on_load'
	/private/var/www/apps/my_project/shared/bundle/ruby/2.7.0/gems/activesupport-6.1.0/lib/active_support/lazy_load_hooks.rb:42:in `each'
	/private/var/www/apps/my_project/shared/bundle/ruby/2.7.0/gems/activesupport-6.1.0/lib/active_support/lazy_load_hooks.rb:42:in `on_load'
	/private/var/www/apps/my_project/shared/bundle/ruby/2.7.0/gems/railties-6.1.0/lib/rails/railtie/configuration.rb:67:in `before_initialize'
	/private/var/www/apps/my_project/shared/bundle/ruby/2.7.0/gems/bugsnag-6.19.0/lib/bugsnag/integrations/railtie.rb:22:in `<class:Railtie>'
	/private/var/www/apps/my_project/shared/bundle/ruby/2.7.0/gems/bugsnag-6.19.0/lib/bugsnag/integrations/railtie.rb:11:in `<module:Bugsnag>'
	/private/var/www/apps/my_project/shared/bundle/ruby/2.7.0/gems/bugsnag-6.19.0/lib/bugsnag/integrations/railtie.rb:10:in `<main>'
	/private/var/www/apps/my_project/shared/bundle/ruby/2.7.0/gems/bootsnap-1.7.2/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `require'
	/private/var/www/apps/my_project/shared/bundle/ruby/2.7.0/gems/bootsnap-1.7.2/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `block in require_with_bootsnap_lfi'
	/private/var/www/apps/my_project/shared/bundle/ruby/2.7.0/gems/bootsnap-1.7.2/lib/bootsnap/load_path_cache/loaded_features_index.rb:92:in `register'
	/private/var/www/apps/my_project/shared/bundle/ruby/2.7.0/gems/bootsnap-1.7.2/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in `require_with_bootsnap_lfi'
	/private/var/www/apps/my_project/shared/bundle/ruby/2.7.0/gems/bootsnap-1.7.2/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:31:in `require'
	/private/var/www/apps/my_project/shared/bundle/ruby/2.7.0/gems/zeitwerk-2.4.2/lib/zeitwerk/kernel.rb:34:in `require'
	/private/var/www/apps/my_project/shared/bundle/ruby/2.7.0/gems/bugsnag-6.19.0/lib/bugsnag.rb:202:in `load_integrations'
	/private/var/www/apps/my_project/shared/bundle/ruby/2.7.0/gems/bugsnag-6.19.0/lib/bugsnag.rb:431:in `<main>'
	/private/var/www/apps/my_project/shared/bundle/ruby/2.7.0/gems/bootsnap-1.7.2/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `require'
	/private/var/www/apps/my_project/shared/bundle/ruby/2.7.0/gems/bootsnap-1.7.2/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `block in require_with_bootsnap_lfi'
	/private/var/www/apps/my_project/shared/bundle/ruby/2.7.0/gems/bootsnap-1.7.2/lib/bootsnap/load_path_cache/loaded_features_index.rb:92:in `register'
	/private/var/www/apps/my_project/shared/bundle/ruby/2.7.0/gems/bootsnap-1.7.2/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in `require_with_bootsnap_lfi'
	/private/var/www/apps/my_project/shared/bundle/ruby/2.7.0/gems/bootsnap-1.7.2/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:31:in `require'
	/private/var/www/apps/my_project/shared/bundle/ruby/2.7.0/gems/zeitwerk-2.4.2/lib/zeitwerk/kernel.rb:34:in `require'
	/Users/user/work/projects/my_project/_temp/fetcher.rb:3:in `<top (required)>'
```

**Comment by xander-jones** [comment]
Hi @xxx44yyy, as per https://github.com/bugsnag/bugsnag-ruby/issues/555#issuecomment-805086322, please can you let us know more detail about what's happening in the run up to the error, and include any Bugsnag configuration you're using.

**Comment by mattdyoung** [comment]
@xxx44yyy 
Closing this issue as there aren't enough details for us to reproduce, but if you're able to let us know more details we'd be happy to re-open and investigate.

**Comment by bisrael** [comment]
@mattdyoung @xander-jones -- getting this same error in v6.20; adding bugsnag to Mastodon's source and trying to deploy on Heroku.

I bisected the gem version pinning the bugsnag gem to `'~> 6.10.0'` removes the error here.
v6.11.0+ fails.

Code is on this branch: https://github.com/bisrael/mastodon/commits/shalomsocial-staging

My bugsnag initializer (`config/initializers/bugsnag.rb`) is

```ruby
if ENV['BUGSNAG_API_KEY']
  require 'bugsnag'

  Bugsnag.configure do |config|
    config.api_key = ENV['BUGSNAG_API_KEY']
    config.app_version = ENV['HEROKU_RELEASE_VERSION'] || ENV['RELEASE']
    config.notify_release_stages = ['production', 'staging']
    config.release_stage = ENV['RELEASE_STAGE'] || 'development'

    config.meta_data_filters += ['card_number', 'card_code', 'cardNumber', 'cardCode', 'cvc']
  end
end

```

```shell
-----> Preparing app for Rails asset pipeline
       Running: rake assets:precompile
       rake aborted!
       NoMethodError: undefined method `event_subscription' for Bugsnag::Railtie:Class
       /tmp/build_98c4b429/vendor/bundle/ruby/2.7.0/gems/railties-5.2.4.4/lib/rails/railtie.rb:192:in `method_missing'
       /tmp/build_98c4b429/vendor/bundle/ruby/2.7.0/gems/bugsnag-6.20.0/lib/bugsnag/integrations/railtie.rb:44:in `block (2 levels) in <class:Railtie>'
       /tmp/build_98c4b429/vendor/bundle/ruby/2.7.0/gems/bugsnag-6.20.0/lib/bugsnag/integrations/railtie.rb:44:in `each'
       /tmp/build_98c4b429/vendor/bundle/ruby/2.7.0/gems/bugsnag-6.20.0/lib/bugsnag/integrations/railtie.rb:44:in `block in <class:Railtie>'
       /tmp/build_98c4b429/vendor/bundle/ruby/2.7.0/gems/activesupport-5.2.4.4/lib/active_support/lazy_load_hooks.rb:69:in `block in execute_hook'
       /tmp/build_98c4b429/vendor/bundle/ruby/2.7.0/gems/activesupport-5.2.4.4/lib/active_support/lazy_load_hooks.rb:62:in `with_execution_control'
       /tmp/build_98c4b429/vendor/bundle/ruby/2.7.0/gems/activesupport-5.2.4.4/lib/active_support/lazy_load_hooks.rb:67:in `execute_hook'
       /tmp/build_98c4b429/vendor/bundle/ruby/2.7.0/gems/activesupport-5.2.4.4/lib/active_support/lazy_load_hooks.rb:43:in `block in on_load'
       /tmp/build_98c4b429/vendor/bundle/ruby/2.7.0/gems/activesupport-5.2.4.4/lib/active_support/lazy_load_hooks.rb:42:in `each'
       /tmp/build_98c4b429/vendor/bundle/ruby/2.7.0/gems/activesupport-5.2.4.4/lib/active_support/lazy_load_hooks.rb:42:in `on_load'
       /tmp/build_98c4b429/vendor/bundle/ruby/2.7.0/gems/railties-5.2.4.4/lib/rails/railtie/configuration.rb:66:in `before_initialize'
       /tmp/build_98c4b429/vendor/bundle/ruby/2.7.0/gems/bugsnag-6.20.0/lib/bugsnag/integrations/railtie.rb:22:in `<class:Railtie>'
       /tmp/build_98c4b429/vendor/bundle/ruby/2.7.0/gems/bugsnag-6.20.0/lib/bugsnag/integrations/railtie.rb:11:in `<module:Bugsnag>'
       /tmp/build_98c4b429/vendor/bundle/ruby/2.7.0/gems/bugsnag-6.20.0/lib/bugsnag/integrations/railtie.rb:10:in `<top (required)>'
       /tmp/build_98c4b429/vendor/bundle/ruby/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `require'
       /tmp/build_98c4b429/vendor/bundle/ruby/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `block in require_with_bootsnap_lfi'
       /tmp/build_98c4b429/vendor/bundle/ruby/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache/loaded_features_index.rb:92:in `register'
       /tmp/build_98c4b429/vendor/bundle/ruby/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in `require_with_bootsnap_lfi'
       /tmp/build_98c4b429/vendor/bundle/ruby/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:31:in `require'
       /tmp/build_98c4b429/vendor/bundle/ruby/2.7.0/gems/activesupport-5.2.4.4/lib/active_support/dependencies.rb:291:in `block in require'
       /tmp/build_98c4b429/vendor/bundle/ruby/2.7.0/gems/activesupport-5.2.4.4/lib/active_support/dependencies.rb:257:in `load_dependency'
       /tmp/build_98c4b429/vendor/bundle/ruby/2.7.0/gems/activesupport-5.2.4.4/lib/active_support/dependencies.rb:291:in `require'
       /tmp/build_98c4b429/vendor/bundle/ruby/2.7.0/gems/bugsnag-6.20.0/lib/bugsnag.rb:202:in `load_integrations'
       /tmp/build_98c4b429/vendor/bundle/ruby/2.7.0/gems/bugsnag-6.20.0/lib/bugsnag.rb:431:in `<top (required)>'
       /tmp/build_98c4b429/vendor/bundle/ruby/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `require'
       /tmp/build_98c4b429/vendor/bundle/ruby/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `block in require_with_bootsnap_lfi'
       /tmp/build_98c4b429/vendor/bundle/ruby/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache/loaded_features_index.rb:92:in `register'
       /tmp/build_98c4b429/vendor/bundle/ruby/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in `require_with_bootsnap_lfi'
       /tmp/build_98c4b429/vendor/bundle/ruby/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:31:in `require'
       /tmp/build_98c4b429/vendor/bundle/ruby/2.7.0/gems/activesupport-5.2.4.4/lib/active_support/dependencies.rb:291:in `block in require'
       /tmp/build_98c4b429/vendor/bundle/ruby/2.7.0/gems/activesupport-5.2.4.4/lib/active_support/dependencies.rb:257:in `load_dependency'
       /tmp/build_98c4b429/vendor/bundle/ruby/2.7.0/gems/activesupport-5.2.4.4/lib/active_support/dependencies.rb:291:in `require'
       /tmp/build_98c4b429/config/initializers/bugsnag.rb:2:in `<top (required)>'
       /tmp/build_98c4b429/vendor/bundle/ruby/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:59:in `load'
       /tmp/build_98c4b429/vendor/bundle/ruby/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:59:in `load'
       /tmp/build_98c4b429/vendor/bundle/ruby/2.7.0/gems/activesupport-5.2.4.4/lib/active_support/dependencies.rb:285:in `block in load'
       /tmp/build_98c4b429/vendor/bundle/ruby/2.7.0/gems/activesupport-5.2.4.4/lib/active_support/dependencies.rb:257:in `load_dependency'
       /tmp/build_98c4b429/vendor/bundle/ruby/2.7.0/gems/activesupport-5.2.4.4/lib/active_support/dependencies.rb:285:in `load'
       /tmp/build_98c4b429/vendor/bundle/ruby/2.7.0/gems/railties-5.2.4.4/lib/rails/engine.rb:663:in `block in load_config_initializer'
       /tmp/build_98c4b429/vendor/bundle/ruby/2.7.0/gems/activesupport-5.2.4.4/lib/active_support/notifications.rb:170:in `instrument'
       /tmp/build_98c4b429/vendor/bundle/ruby/2.7.0/gems/railties-5.2.4.4/lib/rails/engine.rb:662:in `load_config_initializer'
       /tmp/build_98c4b429/vendor/bundle/ruby/2.7.0/gems/railties-5.2.4.4/lib/rails/engine.rb:620:in `block (2 levels) in <class:Engine>'
       /tmp/build_98c4b429/vendor/bundle/ruby/2.7.0/gems/railties-5.2.4.4/lib/rails/engine.rb:619:in `each'
       /tmp/build_98c4b429/vendor/bundle/ruby/2.7.0/gems/railties-5.2.4.4/lib/rails/engine.rb:619:in `block in <class:Engine>'
       /tmp/build_98c4b429/vendor/bundle/ruby/2.7.0/gems/railties-5.2.4.4/lib/rails/initializable.rb:32:in `instance_exec'
       /tmp/build_98c4b429/vendor/bundle/ruby/2.7.0/gems/railties-5.2.4.4/lib/rails/initializable.rb:32:in `run'
       /tmp/build_98c4b429/vendor/bundle/ruby/2.7.0/gems/railties-5.2.4.4/lib/rails/initializable.rb:61:in `block in run_initializers'
       /tmp/build_98c4b429/vendor/bundle/ruby/2.7.0/gems/railties-5.2.4.4/lib/rails/initializable.rb:50:in `each'
       /tmp/build_98c4b429/vendor/bundle/ruby/2.7.0/gems/railties-5.2.4.4/lib/rails/initializable.rb:50:in `tsort_each_child'
       /tmp/build_98c4b429/vendor/bundle/ruby/2.7.0/gems/railties-5.2.4.4/lib/rails/initializable.rb:60:in `run_initializers'
       /tmp/build_98c4b429/vendor/bundle/ruby/2.7.0/gems/railties-5.2.4.4/lib/rails/application.rb:361:in `initialize!'
       /tmp/build_98c4b429/config/environment.rb:5:in `<top (required)>'
       /tmp/build_98c4b429/vendor/bundle/ruby/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `require'
       /tmp/build_98c4b429/vendor/bundle/ruby/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `block in require_with_bootsnap_lfi'
       /tmp/build_98c4b429/vendor/bundle/ruby/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache/loaded_features_index.rb:92:in `register'
       /tmp/build_98c4b429/vendor/bundle/ruby/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in `require_with_bootsnap_lfi'
       /tmp/build_98c4b429/vendor/bundle/ruby/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:31:in `require'
       /tmp/build_98c4b429/vendor/bundle/ruby/2.7.0/gems/activesupport-5.2.4.4/lib/active_support/dependencies.rb:291:in `block in require'
       /tmp/build_98c4b429/vendor/bundle/ruby/2.7.0/gems/activesupport-5.2.4.4/lib/active_support/dependencies.rb:257:in `load_dependency'
       /tmp/build_98c4b429/vendor/bundle/ruby/2.7.0/gems/activesupport-5.2.4.4/lib/active_support/dependencies.rb:291:in `require'
       /tmp/build_98c4b429/vendor/bundle/ruby/2.7.0/gems/railties-5.2.4.4/lib/rails/application.rb:337:in `require_environment!'
       /tmp/build_98c4b429/vendor/bundle/ruby/2.7.0/gems/railties-5.2.4.4/lib/rails/application.rb:520:in `block in run_tasks_blocks'
       /tmp/build_98c4b429/vendor/bundle/ruby/2.7.0/gems/sprockets-rails-3.2.2/lib/sprockets/rails/task.rb:61:in `block (2 levels) in define'
       Tasks: TOP => environment
       (See full trace by running task with --trace)
```

**Comment by mattdyoung** [comment]
Thanks for the details @bisrael. We'll take a look and try to figure out what's causing this.

**Comment by yousif-bugsnag** [comment]
This is now fixed in [v6.21.0](https://github.com/bugsnag/bugsnag-ruby/releases/tag/v6.21.0).
